UniWatch/Controllers/StudentController.cs

```diff
@@ -188,5 +189,75 @@ public ActionResult UnEnrollCofirmed(int classId, int studentId)
 
             return RedirectToAction("Index", new { classId = classId });
         }
+
+        /// <summary>
+        /// Displays a page to upload images of the student's face
+        /// </summary>
+        /// <param name="studentId">The id of the student</param>
+        [HttpGet]
+        public ActionResult Upload(int classId, int studentId)
+        {
+            var student = _dataAccess.UserManager.GetStudentById(studentId);
+            if (student == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+            ViewBag.ClassId = classId;
+
+            return View(student);
+        }
+
+        /// <summary>
+        /// Upload images of the student's face
+        /// </summary>
+        /// <param name="studentId">The id of the student</param>
+        /// <param name="files">The images to upload</param>
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Upload(int classId, int studentId, IEnumerable<HttpPostedFileBase> files)
+        {
+            var student = _dataAccess.UserManager.GetStudentById(studentId);
+            if (student == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+
+            if (!files.Any())
```

Comment:
I hit the upload button without selecting any images and this was bypassed. The list had a single null entry so the condition was false. Can you update this and LectureController as well. i.e. if you don't select an image, files.Count == 1 and files.ElementAt(0) == null. It's supposed to send you back to that page with the error "No file selected"